### PR TITLE
Check docs.rs metadata in CI

### PR DESCRIPTION
**Pull request by BD103**
# Objective

- I have a [list of unstable rustdoc features](https://github.com/orgs/bevyengine/projects/16) that I play to start enabling.
- These features need to be enabled in the `[package.metadata.docs.rs]` table of all crates that use them, as well as in CI (see #13173 for related work).
- CI can be used to ensure that these fields are consistent for all crates.

## Solution

- Create `check_docsrs_metadata.sh`, which verifies that all public crates have the correct metadata using `cargo metadata`.
- Run this script in CI.
- Make all crates scrape examples, to be consistent with the main `bevy` crate.
  - This isn't fantastic because most crates don't have their own examples, but I find this preferable over excluding the main `bevy` crate from checks.

## Testing

- Run `./tools/check_docsrs_metadata.sh` locally. It should either pass or give an error message for all invalid crates.
- The CI check should pass.

## Further Questions

- Should this check be always run, only run when `Cargo.toml` is modified (think `dependencies.yml`), or only in the merge queue?

---

## Changelog

- `[package.metadata.docs.rs]` for all public crates is now verified in CI.

### CODE REVIEW

**Comment by mockersf** [comment]
We re trying to avoid shellscripts to be cross platform. Could you add it to the ci rust script?

**Comment by BD103** [comment]
Example error output ([source](https://github.com/bevyengine/bevy/actions/runs/8989564545/job/24692938754?pr=13275)):

image

**Comment by BD103** [comment]
> We re trying to avoid shellscripts to be cross platform. Could you add it to the ci rust script?

Not easily right now. Our current `ci` structure does not support running generic functions, but instead requires everything to be an `xshell::Cmd`. @Brezak and I are both working on improving this, but that's a few steps away.

When we do finish it, this will be one of the first things ported. (`publish.sh` is another potential candidate.)

**Comment by BD103** [comment]
Speaking of which: #13279.

File: tools/check_docsrs_metadata.sh [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+# `publish = []`. It then only selects the fields that we need: name, path, and docs.rs metadata.
+PUBLIC_CRATES=$(
+    echo $CRATE_METADATA | \
+        jq '[.packages[] | select(.publish != []) | { name, docsrs: .metadata.docs.rs, manifest_path }]'
+)
+
+echo Validating all crates...
+
+INVALID_CRATES=$(echo $PUBLIC_CRATES | jq "[.[] | select(.docsrs != $DOCSRS_METADATA)]")
+# INVALID_CRATES=$(echo $PUBLIC_CRATES | jq "[.[] | select(.name == \"yef\")]")
```
Comment (janhohenheim) [inline tools/check_docsrs_metadata.sh:42]: What's up with this?

**Review by janhohenheim — APPROVED** [review]
Given the current limitations regarding the `ci` tool, this seems alright to me.

**Comment by BD103** [comment]
Closing due to inactivity. If someone want to adopt this, feel free!

**Pull request closed without merging**